traffic_monitor/datareq/crstate.go

```diff
@@ -84,10 +88,58 @@ func filterDirectlyPolledCaches(crstates tc.CRStates) tc.CRStates {
 	return filtered
 }
 
-func srvTRStateSelf(localStates peer.CRStatesThreadsafe, directlyPolledOnly bool) ([]byte, error) {
+func srvTRStateSelf(localStates peer.CRStatesThreadsafe, directlyPolledOnly bool, toData todata.TODataThreadsafe) ([]byte, error) {
 	if !directlyPolledOnly {
-		return tc.CRStatesMarshall(localStates.Get())
+		localStatesC := updateStatusAnycast(localStates, toData)
+		return tc.CRStatesMarshall(localStatesC)
 	}
-	unfiltered := localStates.Get()
+	unfiltered := updateStatusAnycast(localStates, toData)
 	return tc.CRStatesMarshall(filterDirectlyPolledCaches(unfiltered))
 }
+
+func updateStatusAnycast(localStates peer.CRStatesThreadsafe, toData todata.TODataThreadsafe) tc.CRStates {
+	localStatesC := localStates.Get()
+	toDataC := toData.Get()
+
+	for cache, _ := range localStatesC.Caches {
+		if _, ok := toDataC.SameIpServers[cache]; ok {
+			// all server partners must be available if they are in reported state
+			allAvailableV4 := true
+			allAvailableV6 := true
+			allIsAvailable := true
+			for partner, _ := range toDataC.SameIpServers[cache] {
+				if partnerState, ok := localStatesC.Caches[partner]; ok {
+					// a partner host is reported but is marked down for too high traffic or load
+					// this host also needs to be marked down to divert all traffic for their
+					// common anycast ip
+					if strings.Contains(partnerState.Status, "REPORTED") &&
```

Comment:
Why use `string.Contains()`? Traffic Monitor usually compares `CacheStatus`es like this:

https://github.com/apache/trafficcontrol/blob/3713e3b990fbea03c40a580da8e58b81914741b5/traffic_monitor/cache/cache.go#L217